syntax = "proto3";

package PB_MfClassic;
option java_package = "com.flipperdevices.nfc.protobuf";

enum Error {
  None = 0;
  NotPresent = 1;
  Protocol = 2;
  Auth = 3;
  Timeout = 4;
}

enum KeyType {
  KeyTypeA = 0;
  KeyTypeB = 1;
}

message AuthRequest {
  uint32 block = 1;
  bytes key = 2;
  KeyType key_type = 3;
}

message AuthResponse {
  Error error = 1;
  uint32 block = 2;
  bytes key = 3;
  KeyType key_type = 4;
  bytes nt = 5;
  bytes nr = 6;
  bytes ar = 7;
  bytes at = 8;
}

message ReadBlockRequest {
    uint32 block = 1;
    bytes key = 2;
    KeyType key_type = 3;
}

message ReadBlockResponse {
    Error error = 1;
    bytes data = 2;
}
